### PR TITLE
Feature/undo

### PR DESCRIPTION
**Pull request by Steditor**
Part of #249 
Does not yet add a button - where should we place it?

### CODE REVIEW

**Comment by stuikomma** [comment]
Doing a manual modification should clear the redo chain. Right now you can undo things, do something, and then redo the previously undone steps.

**Comment by stuikomma** [comment]
Don't load undo plugin on landingpage.

File: src/plugins/editController/BrushHandler.coffee [hunk 1]
```diff
@@ -91,15 +118,24 @@ class BrushHandler
...
 			piwikTracking.trackEvent 'Editor', 'BrushAction', 'MakeEverythingLego'
-			@editController.rerunLegoPipeline selectedNode
-			brickVis = cachedData.brickVisualization
-			brickVis.updateModifiedVoxels()
-			brickVis.updateVisualization(null, true)
+			do apply = =>
+				@editController.rerunLegoPipeline node
+				brickVis = cachedData.brickVisualization
+				brickVis.updateModifiedVoxels()
+				brickVis.updateVisualization(null, true)
```
Comment (stuikomma) [inline src/plugins/editController/BrushHandler.coffee]: Why do we need this? The scope should be fine when invoking directly in redo.

File: src/plugins/undo/undo.coffee [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	onNodeRemove: (node) =>
+		if node is @currentNode
+			@currentNode = nullNode
+		return
+
+	addTask: (undo, redo) =>
+		@currentNode.getPluginData 'undo'
+		.then ({undoTasks, redoTasks}) ->
+			undoTasks.push {undo, redo}
+			redoTasks = []
```
Comment (stuikomma) [inline src/plugins/undo/undo.coffee]: This does not modify the array saved in the node :wink:

File: src/plugins/editController/BrushHandler.coffee (hunk 1 above)
Comment (Steditor) [inline src/plugins/editController/BrushHandler.coffee]: `apply` is needed for the redo below, but also needs to be executed, hence the `do`

**Comment by stuikomma** [comment]
:ship:

**Pull request merged**